sqlmesh/core/context.py

```diff
@@ -331,6 +331,24 @@ def __init__(
 
         self.path, self.config = t.cast(t.Tuple[Path, C], next(iter(self.configs.items())))
 
+        # Separate loader and configurations for different project types (dbt, native, hybrid)
+        self._dbt_loader: t.Optional[Loader] = None
+        self._sqlmesh_loader: t.Optional[Loader] = None
+        self.dbt_configs: t.Dict[Path, C] = {}
+        self.sqlmesh_configs: t.Dict[Path, C] = {}
+        for path, config in self.configs.items():
+            project_type = c.DBT if config.loader.__name__.lower().startswith(c.DBT) else c.NATIVE
```

Comment:
this seems fairly convoluted, was there a reason you didn't do something like `isinstance(config.loader, DbtLoader)` ? Or make `project_type` a property of the loader?

Reply:
`isinstance` wouldn't work because `config.loader` would resolve as type `abc.ABC`. Changed this to be an attribute of the loader for now